### PR TITLE
feat(wal): Write Ahead Log (WAL) table storage model

### PR DESCRIPTION
**Pull request by ideoma**
This PR enables Write Ahead Log (WAL) table storage mode in default configuration settings. 

Write Ahead Log is **NOT** the default storage mode but this PR allows to create WAL tables by using special syntax, having **WAL** word at the table definition.

```sql
CREATE TABLE a_table (
    x int,
    timestamp ts
) timestamp(ts) partition by DAY WAL
``` 

Table must be partitioned in order to be WAL table. 

It is also possible to convert non-WAL table to WAL and back. These command take effect on server restart (see #2947)

```sql
ALTER TABLE <tablename> SET TYPE WAL;
ALTER TABLE <tablename> SET TYPE BYPASS WAL;
```

The main benefits of WAL tables

- Stable ingestion. Writing to WAL tables do not set pressure on the writer so that ILP writing does not pause when Out Of Order transaction hits the table.
- Simultaneous writing from all interfaces. Issues like #1461 are solved for WAL tables.
- Enables future work on replication. Replication will work with WAL tables only.

It is worth noting that WAL table writing is asynchronous and at the moment there are no sync writing option e.g. when sender or reader can wait when the data is applied to the table.

Here is list of main WAL related pull requests.

#2144 WAL implementation
#2637 Housekeeping WAL segments
#2752 support of DROP and RENAME of WAL tables
#2790 WAL ILP implementation
#2922 optimise WAL application (merge) to the table

### CODE REVIEW

File: core/src/main/resources/io/questdb/site/conf/server.conf [hunk 1]
```diff
@@ -616,6 +616,37 @@ query.timeout.sec=60
...
+################ WAL settings ##################
+
+#Enable support of creating and writing to WAL tables
+#cairo.wal.supported=true
+
+#If set to true WAL becomes the default mode for newly created tables. Impacts table created from ILP and SQL if WAL / BYPASS WAL not specified
+#cairo.wal.enabled.default=false
+
+#If set to true WAL becomes the default mode for newly created tables. Impacts table created from ILP and SQL if WAL / BYPASS WAL not specified
+#cairo.wal.enabled.default=false
```
Comment (marregui) [inline core/src/main/resources/io/questdb/site/conf/server.conf:628]: repeated block, comment and property definition

**Review by marregui — CHANGES_REQUESTED** [review]

File: core/src/main/resources/io/questdb/site/conf/server.conf (hunk 1 above)
Comment (ideoma) [reply core/src/main/resources/io/questdb/site/conf/server.conf]: removed

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|

**Review by marregui — DISMISSED** [review]
LGTM

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|

**Review by glasstiger — APPROVED** [review]

**Pull request merged**